cleanlab/segmentation/filter.py

```diff
@@ -0,0 +1,210 @@
+# Copyright (C) 2017-2023  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+
+"""
+Methods to find label issues in semantic segmentation datasets, where each pixel in a image receives its own class label.
+
+"""
+
+from cleanlab.experimental.label_issues_batched import find_label_issues_batched
+import numpy as np
+from typing import Tuple
+
+
+def find_label_issues(
+    labels: np.ndarray,
+    pred_probs: np.ndarray,
+    *,
+    downsample: int = 16,
+    batch_size: int = 10000,
+    verbose: bool = True,
+    **kwargs,
+) -> np.ndarray:
+    """
+    Returns a boolean mask for the entire dataset, per pixel where ``True`` represents
+    an example identified with a label issue and ``False`` represents an example of a pixel correctly labeled.
+
+    N - Number of images in the dataset
+    K - Number of classes in the dataset
+    H - Height of each image
+    W - Width of each image
+
+    Tip: if you encounter the error "pred_probs is not defined", try setting
+    ``n_jobs=1``.
+
+    Parameters
+    ----------
+    labels : np.ndarray
+      A discrete array of shape ``(N,H,W,)`` of noisy labels for a classification dataset, i.e. some labels may be erroneous.
+      *Format requirements*: for a dataset with K classes, each pixel must be integer in 0, 1, ..., K-1.
+
+    Tip: If your labels are one hot encoded you can `np.argmax(labels_one_hot,axis=1)` assuming that `labels_one_hot` is of dimension (N,K,H,W)
+    before entering in the function
+
+    pred_probs : np.ndarray
+      An array of shape ``(N,K,H,W,)`` of model-predicted class probabilities,
+      ``P(label=k|x)``. Each pixel contains an array of K classes, where for
+      an example `x` the array at each pixel contains the model-predicted probabilities
+      that `x` belongs to each of the K classes. The columns must be ordered such that these probabilities
+      correspond to class 0, 1, ..., K-1.
+
+    downsample : int, optional
+      Factor to shrink labels and pred_probs by. Default ``16``
+      Must be a factor divisible by both the labels and the pred_probs. Note that larger factors result in a linear
+      decrease in performance
```

Comment:
what is "linear decrease in performance"?

That language sounds quite concerning to a user. 
I'd reword or instead say something like:

Larger values of `downsample` produce faster runtimes but potentially less accurate results due to over-compression. Set = 1 to avoid any downsampling.